extensions/scarb-snforge-test-collector/src/crate_collection.rs

```diff
@@ -0,0 +1,82 @@
+use anyhow::{anyhow, Context, Result};
+use camino::{Utf8Path, Utf8PathBuf};
+use serde::Serialize;
+use walkdir::WalkDir;
+
+#[derive(Debug, PartialEq, Clone, Serialize)]
+pub enum CrateLocation {
+    /// Main crate in a package
+    Lib,
+    /// Crate in the `tests/` directory
+    Tests,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct TestCompilationTarget {
+    pub crate_root: Utf8PathBuf,
+    pub crate_name: String,
+    pub crate_location: CrateLocation,
+    pub lib_content: String,
+}
+
+pub fn collect_test_compilation_targets(
+    package_path: &Utf8Path,
+    package_name: &str,
+    package_source_dir_path: &Utf8Path,
+) -> Result<Vec<TestCompilationTarget>> {
+    let mut compilation_targets = vec![TestCompilationTarget {
+        crate_root: package_source_dir_path.to_path_buf(),
+        crate_name: package_name.to_string(),
+        crate_location: CrateLocation::Lib,
+        lib_content: std::fs::read_to_string(package_source_dir_path.join("lib.cairo"))?,
+    }];
+
+    let tests_dir_path = package_path.join("tests");
+    if tests_dir_path.exists() {
+        compilation_targets.push(TestCompilationTarget {
+            crate_name: "tests".to_string(),
+            crate_location: CrateLocation::Tests,
+            lib_content: get_or_create_test_lib_content(tests_dir_path.as_path())?,
+            crate_root: tests_dir_path,
+        });
+    }
+
+    Ok(compilation_targets)
+}
+
+fn get_or_create_test_lib_content(tests_folder_path: &Utf8Path) -> Result<String> {
+    let tests_lib_path = tests_folder_path.join("lib.cairo");
+    if tests_lib_path.try_exists()? {
+        return Ok(std::fs::read_to_string(tests_lib_path)?);
```

Comment:
vide supra

Reply:
We can enforce the path here as it is `tests/lib.cairo`, it has nothing to do with package source path. It is an entrypoint to our test target

Reply:
Oh yes, but please add context as FS::read error will not be easy to pin to specific file.